packages/cosmic-swingset/src/chain-main.js

```diff
@@ -236,17 +236,18 @@ export default async function main(progname, args, { env, homedir, agcc }) {
       }
     }
 
-    const vatconfig = new URL(
-      await importMetaResolve(
-        '@agoric/vats/decentral-config.json',
-        import.meta.url,
-      ),
-    ).pathname;
     const argv = {
       ROLE: 'chain',
       noFakeCurrencies: !env.FAKE_CURRENCIES,
       bootMsg,
     };
+    const vatconfig = new URL(
+      await importMetaResolve(
+        env.CHAIN_BOOTSTRAP_VAT_CONFIG ||
+          argv.bootMsg.params.bootstrap_vat_config,
```

Comment:
hm... I sometimes think of `import.meta.resolve` as kinda powerless as it's just used to resolve design-time/build-time data, but adding `env.XXX` drifts further into runtime IO.

Can it be used to load arbitrary runtime data? Or only to choose between an enumeration of configurations, all of which are decided at design-time?

Reply:
> hm... I sometimes think of `import.meta.resolve` as kinda powerless as it's just used to resolve design-time/build-time data, but adding `env.XXX` drifts further into runtime IO.

The resulting file is parsed as JSON, which at least prevents arbitrary evaluation.

> Can it be used to load arbitrary runtime data? Or only to choose between an enumeration of configurations, all of which are decided at design-time?

It can load anything that `import(...)` can point to.